data/selinux/snappy.te

```diff
@@ -503,8 +503,14 @@ userdom_delete_user_tmp_files(snappy_mount_t)
 files_manage_generic_tmp_dirs(snappy_mount_t)
 files_manage_generic_tmp_files(snappy_mount_t)
 
-# Allow snap-{update,discard}-ns to mount on /etc/
-allow snappy_mount_t etc_t:dir mounton;
+# Allow snap-{update,discard}-ns to manipulate /etc/
+allow snappy_mount_t etc_t:dir { mounton write rmdir remove_name };
+
+# Allow snap-{update,discard}-ns to mount /boot/
+gen_require(`
```

Comment:
Are you sure that this `gen_require` is required here? For the use below, it seems like it was used for `:filesystem` (rather than `:dir`) and for when `:dir` was used with `{ add_name remove_name write search };`, rather than the `{ mounton }` which is required here... I'm really not sure, I just didn't see the `gen_require` required for other `:dir { mounton }` uses elsewhere.

Reply:
Yes, the `boot_t` type must be imported